examples/nodejs/index.js

```diff
@@ -0,0 +1,289 @@
+      throw err;
+    }
+  }
+  // Use a TCP socket.
+  return createTcpPool(config);
+};
+
+const ensureSchema = async pool => {
+  const hasTable = await pool.schema.hasTable('votes');
+  if (!hasTable) {
+    return pool.schema.createTable('votes', table => {
+      table.increments('vote_id').primary();
+      table.timestamp('time_cast', 30).notNullable();
+      table.specificType('candidate', 'CHAR(6)').notNullable();
+    });
+  }
+  logger.info("Ensured that table 'votes' exists");
+};
+
+const createPoolAndEnsureSchema = async () =>
+  await createPool()
+    .then(async pool => {
+      await ensureSchema(pool);
+      return pool;
+    })
+    .catch(err => {
+      logger.error(err);
+      throw err;
+    });
+
+// [START alloydb_knex_connection]
+/**
+ * Insert a vote record into the database.
+ *
+ * @param {object} pool The Knex connection object.
+ * @param {object} vote The vote record to insert.
+ * @returns {Promise}
+ */
+const insertVote = async (pool, vote) => {
+  try {
+    return await pool('votes').insert(vote);
+  } catch (err) {
+    throw Error(err);
+  }
+};
+// [END alloydb_knex_connection]
+
+/**
+ * Retrieve the latest 5 vote records from the database.
+ *
+ * @param {object} pool The Knex connection object.
+ * @returns {Promise}
+ */
+const getVotes = async pool => {
+  return await pool
+    .select('candidate', 'time_cast')
+    .from('votes')
+    .orderBy('time_cast', 'desc')
+    .limit(5);
+};
+
+/**
+ * Retrieve the total count of records for a given candidate
+ * from the database.
+ *
+ * @param {object} pool The Knex connection object.
+ * @param {object} candidate The candidate for which to get the total vote count
+ * @returns {Promise}
+ */
+const getVoteCount = async (pool, candidate) => {
+  return await pool('votes').count('vote_id').where('candidate', candidate);
+};
+
+const httpGet = async (req, res) => {
+  pool = pool || (await createPoolAndEnsureSchema());
+  try {
+    // Query the total count of "TABS" from the database.
+    const tabsResult = await getVoteCount(pool, 'TABS');
+    const tabsTotalVotes = parseInt(tabsResult[0].count);
+    // Query the total count of "SPACES" from the database.
+    const spacesResult = await getVoteCount(pool, 'SPACES');
+    const spacesTotalVotes = parseInt(spacesResult[0].count);
+    // Query the last 5 votes from the database.
+    const votes = await getVotes(pool);
+    // Calculate and set leader values.
+    let leadTeam = '';
+    let voteDiff = 0;
+    let leaderMessage = '';
+    if (tabsTotalVotes !== spacesTotalVotes) {
+      if (tabsTotalVotes > spacesTotalVotes) {
+        leadTeam = 'TABS';
+        voteDiff = tabsTotalVotes - spacesTotalVotes;
+      } else {
+        leadTeam = 'SPACES';
+        voteDiff = spacesTotalVotes - tabsTotalVotes;
+      }
+      leaderMessage =
+        `${leadTeam} are winning by ${voteDiff} vote` + voteDiff > 1 ? 's' : '';
```

Comment:
it's a bit awkward to mix template literals with string concatenation, something like this would be a bit more idiomatic: 

```suggestion
        `${leadTeam} are winning by ${voteDiff} vote${voteDiff > 1 ? 's' : ''}`;
```